build-tools/packages/build-cli/src/commands/release/prepare.ts

```diff
@@ -0,0 +1,266 @@
+	// like we do in CI.
+	const result = await execa.command(`npm run policy-check`, {
+		cwd: context.gitRepo.resolvedRoot,
+	});
+
+	if (result.exitCode !== 0) {
+		return {
+			result: false,
+			message: "Policy check failed. These failures must be fixed before release.",
+			fixCommand: "pnpm run policy-check:fix",
+		};
+	}
+
+	return { result: true };
+}
+
+/**
+ * Returns true if there are no asserts to be tagged.
+ */
+async function checkAsserts(context: Context): Promise<CheckResult> {
+	// policy-check is scoped to the path that it's run in. Since we have multiple folders at the root that represent
+	// the client release group, we can't easily scope it to just the client. Thus, we always run it at the root just
+	// like we do in CI.
+	await execa.command(`npm run policy-check:asserts`, {
+		cwd: context.gitRepo.resolvedRoot,
+	});
+
+	// check for policy check violation
+	const afterPolicyCheckStatus = await context.gitRepo.getStatus();
+	if (afterPolicyCheckStatus !== "") {
+		const repo = new Repository({ baseDir: context.repo.resolvedRoot });
+		await repo.gitClient.reset(ResetMode.HARD);
+		return {
+			result: false,
+			message: "Found some untagged asserts. These should be tagged before release.",
+			fixCommand: "pnpm run policy-check:asserts",
+		};
+	}
+
+	return { result: true };
+}
+
+/**
+ * All the checks that should be executed. The checks are executed serially in this order.
+ */
+const allChecks: ReadonlyMap<string, CheckFunction> = new Map([
+	["Branch has no local changes", checkNoLocalChanges],
+	["Has a microsoft/FluidFramework remote", checkHasRemote],
+	["The local branch is up to date with the remote", checkBranchUpToDate],
+	["Dependencies are installed locally", checkDepsInstalled],
+	["Has no pre-release Fluid dependencies", checkHasNoPrereleaseDependencies],
+	["No repo policy violations", checkPolicy],
+	["No untagged asserts", checkAsserts],
+]);
+
+export class ReleasePrepareCommand extends BaseCommand<typeof ReleasePrepareCommand> {
+	static readonly summary =
+		`Runs checks on a local branch to verify it is ready to serve as the base for a release branch.`;
+
+	static readonly description = `Runs the following checks:\n\n- ${[...allChecks.keys()].join(
+		"\n- ",
+	)}`;
+
+	static readonly args = {
+		package_or_release_group: packageOrReleaseGroupArg({
+			description:
+				"The name of a package or a release group. Defaults to the client release group if not specified.",
+			default: "client",
+		}),
+	};
+
+	async run(): Promise<void> {
+		const context = await this.getContext();
+
+		const rgArg = this.args.package_or_release_group;
+		const pkgOrReleaseGroup = findPackageOrReleaseGroup(rgArg, context);
+		if (pkgOrReleaseGroup === undefined) {
+			this.error(`Can't find package or release group "${rgArg}"`, { exit: 1 });
+		}
+		this.verbose(`Release group or package found: ${pkgOrReleaseGroup.name}`);
+
+		this.logHr();
```

Comment:
Unrelated to this PR but I had to look this up. In MS we have lots of code that logs HResults, so log Hr isn't as unambiguous as you might expect. At least the method is documented, but personally I'd prefer avoiding abbreviations. While apparently Hr is a thing from html, this code is pretty from the dom/front end so I think "hogHorizontalRule" would be more clear. Maybe as some point we might want to rename that.

Reply:
Noted for a possible future change.